docs/src/design/changing_the_primal.md

```diff
@@ -0,0 +1,333 @@
+See the [code for this in ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl/blob/v0.7.49/src/rulesets/LinearAlgebra/factorization.jl#L209-L218).
+If we could not compute remember the eigenvectors in the primal pass we would have to call `eigen` in the gradient pass anyway, and fully recomputed eigenvectors and eigenvalues; over doubling the total work.
+
+However, if you trace this down, it actually used a different algorithm.
+
+`eigvals` basically wraps `LAPACK.syevr!('N', ...)`
+which goes through [DSYEVR](http://www.netlib.org/lapack/explore-html/d2/d8a/group__double_s_yeigen_gaeed8a131adf56eaa2a9e5b1e0cce5718.html)
+and eventually calls [DSTERF](http://www.netlib.org/lapack/explore-html/d2/d24/group__aux_o_t_h_e_rcomputational_gaf0616552c11358ae8298d0ac18ac023c.html#gaf0616552c11358ae8298d0ac18ac023c)
+which uses _"Pal-Walker-Kahan variant of the QL or QR algorithm."_ to compute eigenvalues
+
+In contrast, `eigen` wraps `LAPACK.syevr!('V',...)`
+which also goes through [DSYEVR](http://www.netlib.org/lapack/explore-html/d2/d8a/group__double_s_yeigen_gaeed8a131adf56eaa2a9e5b1e0cce5718.html)
+but eventually calls [DSTEMR](http://www.netlib.org/lapack/explore-html/da/dba/group__double_o_t_h_e_rcomputational_ga14daa3ac4e7b5d3712244f54ce40cc92.html#ga14daa3ac4e7b5d3712244f54ce40cc92)
+which calculates eigenvalues _"either by bisection or the dqds algorithm."_.
+
+Both of these are very good algorithms.
+LAPACK has had decades of work by experts and is one of the most trusted libraries for linear algebra.
+But they are different algorithms that give different results.
+The differences in practices are around $10^{-15}$, which while very small on absolute terms
+are as far as `Float64` is concerned a very real difference.
+
+### Matrix Division
+Roughly speaking:
+`Y=A\B` is the function that finds the least-square solution to `YA ≈ B`.
+When solving such a system, the efficient way to do so is to factorize `A` into an appropriate factorized form such as `Cholesky` or `QR` etc,
+then perform the `\` operation on the factorized form.
+The pullback of `A\B` with respect to `B` is `Ȳ-> A' \ Ȳ`.
+It should be noted that that involves computing the factorization of `A'` (the adjoint of `A`).
+In this computation the factorization of the original `A` can reused.
+Doing so can give a 4x speed-up.
+
+We don't have this in ChainRules.jl yet, because Julia is missing some definitions of `adjoint` of factorizations ([JuliaLang/julia#38293](https://github.com/JuliaLang/julia/issues/38293)).
+I have been promised them for Julia v1.7 though.
+You can see what the code would look like in [PR #302](https://github.com/JuliaDiff/ChainRules.jl/pull/302).
+
+
+# Conclusion
+This document has explained why [`rrule`](@ref) is the way it is.
+In particular it has highlighted why the primal computation is able to be changed from simply calling the function.
+Futher, it has explain where `rrule` returns a closure.
```

Comment:
explain->explained, and maybe where->why?

Reply:
```suggestion
Futher, it has explained why `rrule` returns a closure for the pullback, rather than it being a seperate function.
```